src/lib/optimizer/enumerate_ccp.hpp

```diff
@@ -0,0 +1,52 @@
+#pragma once
+
+#include <cstdint>
+#include <optional>
+#include <utility>
+#include <vector>
+
+#include "join_graph_edge.hpp"
+
+namespace opossum {
+
+class JoinGraph;
+
+/**
+ * CsgCmpPair enumeration algorithm described in in https://dl.acm.org/citation.cfm?id=1164207
+ */
+class EnumerateCcp final {
+ public:
+  EnumerateCcp(const size_t num_vertices, std::vector<std::pair<size_t, size_t>> edges);
+
+  // Corresponds to EnumerateCsg in the paper
+  std::vector<std::pair<JoinGraphVertexSet, JoinGraphVertexSet>> operator()();
+
+ private:
```

Comment:
I think in most other header files we first have private methods and then private class variables. Any reason you changed the order here?